qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -328,9 +327,15 @@ def _restore_circuit_job(
             IBMBackendApiProtocolError: If unexpected return value received
                  from the server.
         """
-        job_id = job_info.get("job_id", "")
+        job_params = {
+            "job_id": job_info["id"],
+            "creation_date": job_info["created"],
+            "status": job_info["status"],
+            "runtime_client": self._provider._runtime_client,
+            "tags": job_info.get("tags"),
+        }
         # Recreate the backend used for this job.
-        backend_name = job_info.get("_backend_info", {}).get("name", "unknown")
+        backend_name = job_info.get("backend")
         try:
             backend = self._provider.get_backend(backend_name)
```

Comment:
This should also pass in the h/g/p to get the right backend. The job's h/g/p is returned by the API.